pyfstat/make_sfts.py

```diff
@@ -1378,39 +1378,40 @@ def get_h0(self, t):
         return self.h0
 
     def concatenate_sft_files(self):
-        """Deals with multiple SFT files via lalapps_splitSFTs executable."""
-        SFTFilename = lalpulsar.OfficialSFTFilename(
-            self.detectors[0],
-            self.detectors[1],
-            self.nsfts,
-            self.Tsft,
-            int(self.tstart),
-            int(self.duration),
-            self.label,
+        """Merges the individual SFT files via lalapps_splitSFTs executable."""
+
+        SFTFilename = (
+            f"{self.detectors[0]}-{self.nsfts}_{self.detectors}_{self.Tsft}SFT"
```

Comment:
This line is just beautiful.

I guess `self.detectors` is checked to be a string somewhere?

Reply:
> This line is just beautiful.

For reference, the original lalapps snprintf beauty is https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalapps/src/pulsar/SFTTools/splitSFTs.c#L688

> I guess self.detectors is checked to be a string somewhere?

Purely coincidentally and implicitly, yes:

https://github.com/PyFstat/PyFstat/blob/af6d263c1a8da07e54b3a7649a7fdca855cd2341/pyfstat/make_sfts.py#L1337

Reply:
But actually if I change in the test to `detectors="H1,L1"`, this one does not catch it. Hmm.

Reply:
Ah, because it's only reached if adding fake Alpha,Delta dependent modulation.

Will add an explicit check earlier.

Reply:
done in d1443ddf0e106ca73ca6b44205e52452507635da